beacon-chain/sync/validate_beacon_blocks.go

```diff
@@ -35,10 +37,47 @@ var (
 	errRejectCommitmentLen = errors.New("[REJECT] The length of KZG commitments is less than or equal to the limitation defined in Consensus Layer")
 )
 
+func (s *Service) validateBeaconBlockPubSub(ctx context.Context, pid peer.ID, msg *pubsub.Message) (pubsub.ValidationResult, error) {
+	done := make(chan struct {
+		result pubsub.ValidationResult
+		err    error
+	})
+
+	go func() {
+		result, err := func() (pubsub.ValidationResult, error) {
+			return s.validateBeaconBlockPubSubInternal(ctx, pid, msg)
+		}()
+		done <- struct {
+			result pubsub.ValidationResult
+			err    error
+		}{result, err}
+	}()
+
+	select {
+	case res := <-done:
+		return res.result, res.err
+	case <-time.After(12 * time.Second):
+		filePath := "/tmp/pre-confirm-blocks.goroutine"
+		file, err := os.Create(filePath)
+		if err != nil {
+			log.WithError(err).Error("Could not create file for goroutine dump")
+		}
+		defer func() {
+			if err := file.Close(); err != nil {
+				log.WithError(err).Error("Could not close file for goroutine dump")
+			}
+		}()
+		if err := pprof.Lookup("goroutine").WriteTo(file, 2); err != nil {
+			log.WithError(err).Error("Could not write goroutine dump to file")
+		}
+	}
```

Comment:
This looks like debug/testing code that cannot be disabled or configured. Remove?